### ISSUE TITLE
Unable to delete characters after ZSH edit-command-line ZLE widget

### ISSUE THREAD

**Issue by atreyasha** [alacritty/alacritty#5172]
After using the `edit-command-line` ZLE widget for `zsh`, I can no longer delete the characters that are pasted into the command-line. A quick example is shown below:

![bug](https://user-images.githubusercontent.com/35427332/119369199-3092e300-bcb4-11eb-9d45-a1ba460de7a7.gif)

This issue seems very similar to #2026 and #3219. I tried the mentioned workarounds like specifying the `--login` flag in Alacritty's configuration file and also fully specifying all locales. None of these fixed this problem for me, which is why I opened this issue. It is also worth mentioning that I did not encounter this issue with other terminal emulators such as the now-deprecated `termite`.

Regarding the `edit-command-line` widget, the relevant lines that define it can be found below or in my [dotfiles](https://github.com/atreyasha/dotfiles/blob/a60b11be0c7b00b49d59dceadfd2f523f6a8a46d/zsh/.zsh.d/keybindings#L54-L57):

```shell
# sourced by ~/.zshrc
autoload edit-command-line
zle -N edit-command-line
bindkey '^e' edit-command-line
```

### System

OS: Arch Linux 5.10.38-1-lts
Version: alacritty 0.8.0 (a1b13e68)
Linux/BSD: X11, i3-gaps, picom

**Comment by chrisduerr** [comment]
Could you try a different terminal please? Something up to date like vte/kitty maybe which supports bracketed paste by default.

**Comment by atreyasha** [comment]
Yes, tested with `kitty` and `gnome-terminal` as shown below.

### kitty

Version: 0.20.3
Status: This issue cannot be reproduced

![kitty](https://user-images.githubusercontent.com/35427332/119462776-65994700-bd41-11eb-859e-d58b900e1193.gif)

### gnome-terminal

Version: 3.40.1
Status: This issue can be reproduced

![gnome-terminal](https://user-images.githubusercontent.com/35427332/119462795-69c56480-bd41-11eb-8fdb-b06e19e23401.gif)

**Comment by chrisduerr** [comment]
If a different terminal like gnome-terminal can reproduce this, chances are this is not a terminal problem. It doesn't sound like one anyways.

**Comment by atreyasha** [comment]
I see. FWIW it can't be reproduced with kitty.

> [...] chances are this is not a terminal problem. It doesn't sound like one anyways.

What kind of a problem does it sound like instead?

**Comment by atreyasha** [comment]
In case anyone faces the same issue, I found the cause.

The problem was not reproduced when I started the default `alacritty` instance with `alacritty --config-file /dev/null`, so the cause was the `TERM` environmental variable in my configuration:

```yaml
env:
  TERM: xterm-256color
```

I had set this to circumvent some issues with remote shells based on a section in the Arch [Wiki](https://wiki.archlinux.org/title/Alacritty), but obviously there are disadvantages to doing this.

Removing this definition for `TERM` fixes the issue for me.
